src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -58,6 +59,15 @@ CompilationStartAnalysisContext context
 				SyntaxKind.IdentifierName,
 				SyntaxKind.GenericName
 			);
+
+			context.RegisterSyntaxNodeAction(
+				ctx => AnalyzeConditionalImmutability(
+					ctx,
+					ctx.Node
+				),

```

Comment:
nit: splitting here is fine but we usually spill only when it gets to 80 chars or just there is a lot to read/understand.

less-nit: you can probably drop the second argument here and just get it from ctx inside the method declartion. We pass lambdas as the action argument when we need to inject something from the stack into the action (like `immutabilityContext`). This happens because these analyzers can be reused in parallel for different compilations, so we don't want to store this stuff in a member variable. So what I'm suggesting:

```suggestion
				AnalyzeConditionalImmutability,
```

(but you'd also need to tweak the defn of that method)

Reply:
The pattern in our analyzers of pulling out the node early here is when we're registering for one kind and we can confidently cast the `SyntaxNode` to the type we've registered for.

Reply:
Yeah, but we're not casting here, and can't because it could be one of two things.

Reply:
Other examples:

https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/Language/ClassShouldBeSealedAnalyzer.cs#L25
https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/Language/DefaultValueConsistencyAnalyzer.cs#L31
https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs#L46-L64
https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/Language/StructShouldBeReadonlyAnalyzer.cs#L29